### PR TITLE
[CBRD-24017] The problem that predicate push is not performed without changing the outer join to the inner join.

### PR DESCRIPTION
**Pull request by shparkcubrid**
http://jira.cubrid.org/browse/CBRD-24017

info_spec.appears, info->nullable is not initialized properly. and I fixed it to handle right outer join effectively using mq_is_outer_join_spec().

### CODE REVIEW

**Comment by shparkcubrid** [comment]
8a4c98a bfda1c9 are related to this PR.

**Comment by shparkcubrid** [comment]
sql/_13_issues/_15_1h/cases/bug_bts_13571.sql - linux_sql_64bit_release
sql/_13_issues/_15_1h/cases/bug_bts_7547_1.sql - linux_sql_64bit_release
This is a case that an outer join is changed to an inner join.
In my opinion, it is right to change. Please check it.

File: src/optimizer/query_rewrite.c [hunk 1]
```diff
@@ -5495,27 +5495,21 @@ qo_rewrite_outerjoin (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *c
       prev_spec = NULL;
       for (spec = node->info.query.q.select.from; spec; prev_spec = spec, spec = spec->next)
 	{
-	  if (spec->info.spec.join_type == PT_JOIN_LEFT_OUTER
-	      || (spec->info.spec.join_type == PT_JOIN_RIGHT_OUTER && prev_spec))
+	  /* check outer join spec. */
```
Comment (beyondykk9) [inline src/optimizer/query_rewrite.c:5497]: Currently, prev_spec is necessary ? it looks not useless and the prev_spec was a condition to check OUTER JOIN.
it's no problem that the condtion is changed to check OUTER JOIN without prev_spec ?

Comment (shparkcubrid) [reply src/optimizer/query_rewrite.c:5497]: I removed prev_spec.
Using mq_is_outer_join_spec(), right outer join is checked more effectively.

Comment (beyondykk9) [reply src/optimizer/query_rewrite.c:5497]: OK.

**Review by beyondykk9 — APPROVED** [review]

**Review by ctshim — APPROVED** [review]

**Review by hgryoo — APPROVED** [review]

**Review by youngjinj — APPROVED** [review]

**Pull request merged**